Added minimal RTD configuration file.

carltongibson opened the pull request:
Use of the configuration file will be mandatory from Sept 2023.

https://blog.readthedocs.com/migrate-configuration-v2/

Hey @smithdc1 👋 — I'm not 100% sure this is everything needed, but it's a barebones beginning. LMKWYT

carltongibson committed:
Added minimal RTD configuration file.

Use of the configuration file will be mandatory from Sept 2023.

https://blog.readthedocs.com/migrate-configuration-v2/

bryan-brancotte commented:
Hi @carltongibson 

Was looking at this topic, saw your already started to tacle the issue.

I think we also should add requirements file, I don't think it will be automatically detected otherwise

https://docs.readthedocs.io/en/stable/config-file/v2.html#requirements-file
```yaml
python:
  install:
    - requirements: docs/requirements.txt
```

bryan-brancotte commented:
Ok, I have enabled doc for this branch on rtd, and testing to make it work

bryan-brancotte committed:
specify requirements to install

bryan-brancotte committed:
Follow rtd theme, set sphinx lower than 7.0

As of https://github.com/readthedocs/sphinx_rtd_theme/blob/9899ee4ee2f547f81e51297dc12317f018e62fdd/setup.cfg#L48 Sphinx 7 is not supported, prevent using it

bryan-brancotte committed:
Use recent rtd theme

bryan-brancotte committed:
also build epub and pdf

Following https://docs.readthedocs.io/en/stable/config-file/v2.html#formats

bryan-brancotte committed:
removing limit as should be dictated by sphinx_rtd_theme (#1356)

carltongibson reviewed (commented):
Brilliant. Let's give it a go.
